### PR TITLE
Derived commands

### PR DESCRIPTION
**Pull request by bagnell**
Adds derived commands for 3d tiles, shadows, and OIT.

### CODE REVIEW

File: Source/Renderer/DrawCommand.js [hunk 1]
```diff
@@ -15,6 +19,34 @@ define([
...
+        this._renderState = options.renderState;
+        this._framebuffer = options.framebuffer;
+        this._pass = options.pass;
+        this._executeInClosestFrustum = defaultValue(options.executeInClosestFrustum, false);
+        this._owner = options.owner;
+        this._debugShowBoundingVolume = defaultValue(options.debugShowBoundingVolume, false);
+        this._debugOverlappingFrustums = 0;
+        this._dirty = true;
+
+// TODO: how to invalidate this?  Get/set for properties above?  Make some of them (e.g., vertex array) readonly?
```
Comment (pjcozzi) [inline Source/Renderer/DrawCommand.js]: Can we remove this TODO?  It has the same idea we discussed today.

File: Source/Scene/Scene.js [hunk 2]
```diff
@@ -1105,6 +1107,15 @@ define([
             command.debugOverlappingFrustums = 0;
         }
 
+        if (command._dirty) {
```
Comment (pjcozzi) [inline Source/Scene/Scene.js]: Can we write this instead as
```javascript
if (command._dirty) {
    command._dirty = false;
    // ...
```
It avoids a potential cache miss.

**Comment by pjcozzi** [comment]
What's the deal with the `Scene/TileBoundingBox` test failures in CI: https://travis-ci.org/AnalyticalGraphicsInc/cesium/builds/120984476

**Comment by pjcozzi** [comment]
Just those minor comments.  What is the best branch strategy here?  Merge this into `master`, then `master` into `shadows` and `3d-tiles`, then additional derived commands PRs into `shadows` and `3d-tiles`?

File: Source/Renderer/DrawCommand.js (hunk 1 above)
Comment (bagnell) [inline Source/Renderer/DrawCommand.js]: How are we going to release the resources for the command? Primitives release the resources for each command, but who releases the resources for derived commands?

**Comment by bagnell** [comment]
>What's the deal with the Scene/TileBoundingBox test failures in CI

I'm not sure. The tests pass when run locally.

**Comment by bagnell** [comment]
>Just those minor comments. What is the best branch strategy here? Merge this into master, then master into shadows and 3d-tiles, then additional derived commands PRs into shadows and 3d-tiles?

Yes, that was my plan.

File: Source/Renderer/DrawCommand.js (hunk 1 above)
Comment (pjcozzi) [inline Source/Renderer/DrawCommand.js]: The command can own the derived commands.  In some cases - like in 3D Tiles - they will not have new resources.

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look. The failing tests were fixed in #3811.

**Pull request merged**